feat: migrate `rover config whoami` output to stdout and add json format

scombat opened the pull request:
Hello everyone,

I am open to collaborating with you to improve this project. I welcome feedback and suggestions on how to make the code more efficient and effective. Please let me know if you have any thoughts or ideas for improvement. I understood that not all changes may be accepted or merged due to the `triage` label, and that's perfectly fine. I'm open to feedback and willing to make changes as needed to improve the codebase. 
I would like to add functional tests for common user use cases, as well as for graphs with valid and invalid profiles for each case. However, I am not sure if it is relevant or if you have already implemented functional tests.

## Description

BREAKING CHANGE: This pr fixes #1380 by using a RoverOutput `whoami` variant to print the output of the `rover config whoami` command to stdout instead of stderr. It also adds the capability to print the output as json with the  `--json` flag.
This PR migrates the output of rover config whoami command from stderr to stdout and fixes #1380. Additionally, this PR adds the --json flag to allow users to retrieve the output in JSON format. This change improves the usability of the rover CLI and enhances the user experience.

## Related Issues & discussions
- #1380 Rover config whoami should print to stdout instead of stderr and return JSON data
- #60 Design consistent output printing

### Related PRs
- #68 standardize command return type

## WIP :
- [x] Migrate the output of rover config whoami to stdout using `RoverOutput::ConfigWhoAmIOutput` variant
- [x] Add a table format to improve the readability of the output
- [x] Implement the Display trait for the Actor enum to print the actor type in a human-readable format
- [x] Allow users to retrieve the output in JSON format
- [X] Add unit tests for the new methods
- [X] ~Add functional tests for error cases and different output formats~
- [x] ~Add documentation~

## Checklist
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] ~I have added or updated necessary documentation.~
- [x] I sign the [CLA](https://contribute.apollographql.com/). (scombat)
- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**CONTRIBUTING**](https://github.com/apollographql/rover/blob/main/CONTRIBUTING.md)
- [x] I tested my code with `cargo test --workspace` and all tests passed locally.
- [x] I used `cargo clippy` to check for common mistakes and code style issues.
- [x] I used `cargo fmt --all` to format my code.

## Outputs :
### User Actor : 
#### Human readable output  (Keys are displayed in `Style::WhoAmIKey` green but not in images below) [updated]
![ray-so-export (2)](https://user-images.githubusercontent.com/6457707/231460438-f3f2b35b-932f-4d42-8a5f-db4640d354ad.png)

#### Json format (with `--format json` flag) [updated]
![ray-so-export (3)](https://user-images.githubusercontent.com/6457707/231460541-2c4a13a4-c50e-41a8-9b2e-b041965bb74e.png)


### Graph Actor : 
#### Human readable output  (Keys are displayed in `Style::WhoAmIKey` green but not in images below) [updated]
![ray-so-export](https://user-images.githubusercontent.com/6457707/231460624-f4bf318f-28cb-4866-82f7-f6ccb9f5785d.png)


#### Json format (with `--format json` flag) [updated]
![ray-so-export (4)](https://user-images.githubusercontent.com/6457707/231460746-cd6db391-b986-4e1d-a7a8-25d04130e89b.png)

scombat committed:
fix(whoami)!: use standardized RoverOutput instead of writing to stderr

refactor(whoami): extract conditions in helper methods

to improve tests, lisibility and maintenability

test(whoami): add unit tests for actor type guard

test(whoami): add unit tests for masked or unmasked api key

test(whoami): add unit tests for graph title helper

test(whoami): add unit tests for uniq graph id

test(whoami): add uni tests for user id

scombat committed:
feat(RoverOutput)!: add config whoami ouput

feat(RoverOutput)!: add config whoami json output

feat(RoverOuput): add WhoAmI variant

scombat committed:
chore(WhoAmI): implement display trait to WhoAmI::Actor enum

to improve whoami command output readability

EverlastingBugstopper commented on src/command/output.rs:
```diff
@@ -84,6 +92,38 @@ pub enum RoverOutput {
+
+                if let Some(uniq_graph_id) = uniq_graph_id {
+                    table.add_row(row![
+                        Style::WhoAmIKey.paint("Unique Graph ID"),
```
```suggestion
                        Style::WhoAmIKey.paint("Graph ID"),
```

scombat replied:
Done.

EverlastingBugstopper commented on src/command/output.rs:
```diff
@@ -84,6 +92,38 @@ pub enum RoverOutput {
+                table.add_row(row![Style::WhoAmIKey.paint("Key Type"), key_type]);
+
+                if let Some(graph_title) = graph_title {
+                    table.add_row(row![Style::WhoAmIKey.paint("Graph Title"), graph_title]);
```
let's move this underneath `Graph ID`

scombat replied:
Done.

EverlastingBugstopper commented on src/command/output.rs:
```diff
@@ -36,6 +36,14 @@ use termimad::{crossterm::style::Attribute::Underlined, MadSkin};
+        origin: String,
+        api_key: String,
+        graph_title: Option<String>,
+        uniq_graph_id: Option<String>,
```
```suggestion
        graph_id: Option<String>,
```

EverlastingBugstopper replied:
i'd like to rename this to `graph_id`

scombat replied:
Done.

EverlastingBugstopper reviewed (changes requested):
Requested some small changes

scombat committed:
refactor(output): rename uniq_graph_id struct field to graph_id

in RoverOuput::ConfigWhoAmIOuput variant

refactor(output): move the graph title below the graph ID for diplay

refactor(output): reorder ConfigWhoAmIOutput fields alphabetically

refactor(output): replace "Unique Graph ID" displayed key by "Graph ID"

for the human readable format output.

refactor(output): replace "uniq_graph_id" property name with "graph_id"

for the json format output.

scombat committed:
refactor(whoami): rename get_uniq_graph_id method to get_graph_id

refactor(whoami): rename uniq_graph_id field to graph_id in command output

refactor(whoami): reorder command output fields alphabetically

EverlastingBugstopper reviewed (approved):
Thanks!
